### PR TITLE
Handle h2 vs http/1.1 event ordering differences

### PR DESCRIPTION
**Pull request by yschimke**

### CODE REVIEW

**Comment by yschimke** [comment]
https://github.com/square/okhttp/issues/270

File: okhttp-tests/src/test/java/okhttp3/EventListenerTest.java [hunk 1]
```diff
@@ -103,12 +104,47 @@
     assertEquals(200, response.code());
     response.body().close();
 
+    // ordering of response body end currently differs
+    List<String> expectedEvents;
+    if (response.protocol == Protocol.HTTP_2) {
```
Comment (dave-r12) [inline okhttp-tests/src/test/java/okhttp3/EventListenerTest.java]: Should we break this out into distinct test cases? We could disable HTTP/2 on the client in one case, and enforce it on the other.

File: okhttp-tests/src/test/java/okhttp3/EventListenerTest.java [hunk 2]
```diff
@@ -96,27 +98,58 @@
...
-    enableTlsWithTunnel(false);
-    server.enqueue(new MockResponse()
-        .setBody("abc"));
-
+  private void assertSuccessfulEventOrder() throws IOException {
     Call call = client.newCall(new Request.Builder()
-        .url(server.url("/"))
-        .build());
+            .url(server.url("/"))
+            .build());
```
Comment (swankjesse) [inline okhttp-tests/src/test/java/okhttp3/EventListenerTest.java]: nit: we do +4 on wrapped lines

Comment (swankjesse) [inline okhttp-tests/src/test/java/okhttp3/EventListenerTest.java]: If you’re using IntelliJ, it’s
https://github.com/square/java-code-styles/

Comment (yschimke) [inline okhttp-tests/src/test/java/okhttp3/EventListenerTest.java]: @swankjesse my bad, fresh Intellij install.  Is there an automated way to reformat?

**Pull request merged**